### PR TITLE
Add fluentd-windows prometheus metrics support

### PR DESCRIPTION
**Pull request by coutinhop**
Also a small fix to calico-node-windows prometheus metrics.

## For PR author

- [x] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/fluentd.go [hunk 1]
```diff
@@ -214,6 +215,13 @@ func (c *fluentdComponent) fluentdNodeName() string {
 	return FluentdNodeName
 }
 
+func (c *fluentdComponent) fluentdMetricsService() string {
```
Comment (caseydavenport) [inline pkg/render/fluentd.go:218]: Should be `fluentdMetricsServiceName` to be more clear

Comment (caseydavenport) [reply pkg/render/fluentd.go:218]: This should also have a comment explaining why we need / want two names

File: pkg/render/monitor/monitor.go [hunk 2]
```diff
@@ -743,7 +743,13 @@ func (mc *monitorComponent) serviceMonitorCalicoNode() *monitoringv1.ServiceMoni
 			Labels:    map[string]string{"team": "network-operators"},
 		},
 		Spec: monitoringv1.ServiceMonitorSpec{
-			Selector:          metav1.LabelSelector{MatchLabels: map[string]string{"k8s-app": "calico-node"}},
+			Selector: metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{Key: "k8s-app",
```
Comment (caseydavenport) [inline pkg/render/monitor/monitor.go:748]: The indentation here looks a little odd to me. I would expect it to look more like this:

```go
{
  Key: "k8s-app",
  Operator: metav1.LabelSelectorOpIn,
  Values:   []string{"calico-node", "calico-node-windows"},
},
```

File: pkg/render/fluentd.go (hunk 1 above)
Comment (coutinhop) [reply pkg/render/fluentd.go:218]: Will do, thanks!

File: pkg/render/monitor/monitor.go (hunk 2 above)
Comment (coutinhop) [reply pkg/render/monitor/monitor.go:748]: Yes, it's definitely weird, fill fix

File: pkg/render/monitor/monitor_test.go [hunk 3]
```diff
@@ -444,8 +449,13 @@ var _ = Describe("monitor rendering tests", func() {
 		Expect(ok).To(BeTrue())
 		Expect(servicemonitorObj.ObjectMeta.Labels).To(HaveLen(1))
 		Expect(servicemonitorObj.ObjectMeta.Labels["team"]).To(Equal("network-operators"))
-		Expect(servicemonitorObj.Spec.Selector.MatchLabels).To(HaveLen(1))
-		Expect(servicemonitorObj.Spec.Selector.MatchLabels["k8s-app"]).To(Equal("calico-node"))
+		Expect(servicemonitorObj.Spec.Selector.MatchLabels).To(HaveLen(0))
+		Expect(servicemonitorObj.Spec.Selector.MatchExpressions).To(HaveLen(1))
+		Expect(servicemonitorObj.Spec.Selector.MatchExpressions).To(ConsistOf([]metav1.LabelSelectorRequirement{
+			{Key: "k8s-app",
```
Comment (caseydavenport) [inline pkg/render/monitor/monitor_test.go:455]: Same indentation issues down here

**Review by caseydavenport — APPROVED** [review]

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**